test/fork/migration-v2.js

```diff
@@ -886,6 +887,74 @@ describe('V2 upgrade', function () {
+    const { setProductTypesTransaction, setProductsTransaction, productTypeData, productData, productTypeIds } =
+      await generateV2ProductTxs(ethers.provider, this.cover.address, signerAddress);
+
+    console.log('Calling setProductTypes.');
+    await this.abMembers[0].sendTransaction(setProductTypesTransaction);
+
+    console.log('Calling setProducts.');
+    await this.abMembers[0].sendTransaction(setProductsTransaction);
+
+    const productCount = await this.cover.productsCount();
+
+    assert.equal(productCount.toNumber(), productData.length);
+
+    const productTypesCount = await this.cover.productTypesCount();
+    assert.equal(productTypesCount.toNumber(), productTypeData.length);
+
+    for (let i = 0; i < productTypeData.length; i++) {
+      const productType = await this.cover.productTypes(i);
+      const productTypeName = await this.cover.productTypeNames(i);
+
+      const data = productTypeData[i];
+
+      expect(productTypeName).to.be.equal(data.Name);
+      expect(productType.claimMethod.toString()).to.be.equal(data['Claim Method']);
+      expect(productType.gracePeriod.toString()).to.be.equal(data['Grace Period (days)']);
+    }
+
+    const products = await this.cover.getProducts();
+    for (let i = 0; i < products.length; i++) {
+      const product = products[i];
+      const data = productData[i];
+
+      expect(product.productType).to.be.equal(productTypeIds[data['Product Type']]);
+      expect(product.yieldTokenAddress).to.be.equal(
+        data['Product Type'] === 'Yield Token'
+          ? data['Yield Token Address']
+          : '0x0000000000000000000000000000000000000000',
+      );
+
+      const coverAssetsAsText = data['Cover Assets'];
+      const coverAssets =
+        (coverAssetsAsText === 'DAI' && 0b10) || // Yield token cover that uses DAI
+        (coverAssetsAsText === 'ETH' && 0b01) || // Yield token cover that uses ETH
+        0; // The default is 0 - this means all assets are allowed (no whitelist)
+      expect(product.coverAssets.toString()).to.be.equal(coverAssets.toString());
+      expect(product.initialPriceRatio).to.be.equal(parseInt(data['Initial Price Ratio']) * 100);
+      expect(product.capacityReductionRatio).to.be.equal(parseInt(data['Capacity Reduction Ratio']));
+      expect(product.isDeprecated).to.be.equal(false);
+      expect(product.useFixedPrice).to.be.equal(data['Use Fixed Price'] === 'Yes');
+    }
+  });
+
+  it('Migrates existing FTX covers to V2', async function () {
+    const ftxCoverIds = [7907, 7881, 7863, 7643, 7598, 7572, 7542, 7313, 7134];
+
+    for (const coverId of ftxCoverIds) {
+      const { memberAddress } = await this.gateway.getCover(coverId);
+
+      const member = await getSigner(memberAddress);
+      await evm.impersonate(memberAddress);
+      await evm.setBalance(memberAddress, parseEther('1000'));
+
+      await this.coverMigrator.connect(member).migrateCovers([coverId], memberAddress);
```

Comment:
@danoctavian we should assert here that all the fields of the new cover are populated as expected and a claim can be submitted; I'd use the `migrateCoverAndSubmitClaim` function directly